### PR TITLE
refactor inheritance graph

### PR DESCRIPTION
**Pull request by MathisGD**
It makes it at tree, and avoid the need for the _udpate override

### CODE REVIEW

**Review by QGarchery — APPROVED** [review]
Yes, I was thinking the same thing, thank you :heart:

**Review by Jean-Grimal — COMMENTED** [review]
Personally I would only keep the delegation logic in this contract

**Comment by QGarchery** [comment]
> Personally I would only keep the delegation logic in this contract

This is what this PR does, no ? Or you mean that you wouldn't inherit from ERC20Upgradeable ? But I think you have to, because they are linked in the update function

**Review by colin-morpho — APPROVED** [review]

File: src/ERC20DelegatesUpgradeable.sol [hunk 1]
```diff
@@ -18,11 +16,9 @@ import {Initializable} from "../lib/openzeppelin-contracts-upgradeable/contracts
 /// This extension keeps track of the current voting power delegated to each account. Voting power can be delegated
 /// either by calling the `delegate` function directly, or by providing a signature to be used with `delegateBySig`.
 ///
-/// This enables onchain votes on external voting smart contracts leveraging storage proofs.
-///
 /// By default, token balance does not account for voting power. This makes transfers cheaper. Whether an account
 /// has to self-delegate to vote depends on the voting contract implementation.
-abstract contract ERC20DelegatesUpgradeable is Initializable, ERC20Upgradeable, EIP712Upgradeable, IDelegates {
+abstract contract ERC20DelegatesUpgradeable is ERC20PermitUpgradeable, IDelegates {
```
Comment (MerlinEgalite) [inline src/ERC20DelegatesUpgradeable.sol:21]:
```suggestion
abstract contract ERC20PermitDelegatesUpgradeable is ERC20PermitUpgradeable, IDelegates {
```

Should we?

Comment (MerlinEgalite) [reply src/ERC20DelegatesUpgradeable.sol:21]: Ugly but more insightful

Comment (colin-morpho) [reply src/ERC20DelegatesUpgradeable.sol:21]: I figure this would be more consistent.

**Review by colin-morpho — APPROVED** [review]

**Review by QGarchery — DISMISSED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Review by MerlinEgalite — DISMISSED** [review]

**Review by QGarchery — APPROVED** [review]

**Review by colin-morpho — DISMISSED** [review]

**Review by Jean-Grimal — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Review by QGarchery — APPROVED** [review]

**Review by colin-morpho — APPROVED** [review]

**Pull request merged**